portality/view/admin.py

```diff
@@ -144,6 +147,83 @@ def article_endpoint(article_id):
     return resp
 
+def _validate_authors(form, require=1):
+    counted = 0
+    for entry in form.authors.entries:
+        name = entry.data.get("name")
+        if name is not None and name != "":
+            counted += 1
+    return counted >= require
+
+@blueprint.route("/article/<article_id>", methods=["GET", "POST"])
+@login_required
+@ssl_required
+@write_required()
+def article_page(article_id):
+    if not current_user.has_role("edit_article"):
+        abort(401)
+    ap = models.Article.pull(article_id)
+    if ap is None:
+        abort(404)
+
+    form = ArticleForm(id=article_id)
+    form_context = formcontext.ManEdArticleReview(ap)
+
+    if request.method == "GET":
+        return render_template("admin/edit_article_metadata.html", form=form, source=ap, form_context=form_context)
+
+    elif request.method == "POST":
+        form = ArticleForm(request.form, id=article_id, method="post")
+        # first we need to do any server-side form modifications which
+        # the user might request by pressing the add/remove authors buttons
+        more_authors = request.values.get("more_authors")
+        remove_author = None
+        for v in list(request.values.keys()):
+            if v.startswith("remove_authors"):
+                remove_author = v.split("-")[1]
+
+        # if the user wants more authors, add an extra entry
+        if more_authors:
+            form.authors.append_entry()
+            return render_template("admin/edit_article_metadata.html", form=form, form_context=form_context)
+
+        # if the user wants to remove an author, do the various back-flips required
+        if remove_author is not None:
+            keep = []
+            while len(form.authors.entries) > 0:
+                entry = form.authors.pop_entry()
+                if entry.short_name == "authors-" + remove_author:
+                    break
+                else:
+                    keep.append(entry)
+            while len(keep) > 0:
+                form.authors.append_entry(keep.pop().data)
+            return render_template("admin/edit_article_metadata.html", form=form, form_context=form_context)
+
+        # if we get to here, then this is the full submission, and we need to
+        # validate and return
+        enough_authors = _validate_authors(form)
+        if form.validate():
+            # if the form validates, then we have to do our own bit of validation,
+            # which is to check that there is at least one author supplied
+            if not enough_authors:
+                return render_template("admin/edit_article_metadata.html", form=form, author_error=True, form_context=form_context)
+            else:
+                xwalk = ArticleFormXWalk()
+                art = xwalk.crosswalk_form(form, id=article_id)
+                articleService = DOAJ.articleService()
+                try:
+                    articleService.create_article(art, current_user._get_current_object(), add_journal_info=True)
```

Comment:
If we're going to use a FormContext, all this stuff should be in there rather than in the view